mono/mini/mini.h

```diff
@@ -53,9 +53,11 @@
  * The mini code should not have any compile time dependencies on the GC being used, so the same object file from mini/
  * can be linked into both mono and mono-sgen.
  */
+#if !defined(MONO_DLL_EXPORT) || !defined(_MSC_VER)
```

Comment:
Why is this needed ?

Reply:
sgen have some exports in the source files that needs to be exported from dynamic mono library. Since we build all needed sources when building the libmono-dynamic version, we need to set sgen defines for the libmono-dynamic build as well. I didn't want to add defines/file since that will cause maintenance problems so the sgen defines are set for the complete build of libmono-dynamic. The above check then kicks in but is incorrect for this scenario since the build will only use the produced object files when linking libmono-dynamic using specified GC and the object files won't be shared outside the build.

The root problem is the exports from sgen sources, so if we could get away from exporting them from sgen sources and do something similar as already done with other GC exports, we could reuse the objects build in the static version of sgen GC (as we do in the Boehm use case), those objects won't have any __declspec(dllexport) in them since they are build for the static use case.